### PR TITLE
Implement prefix of image urls with a filter

### LINKED ISSUE #820: 🐛 [Bug] Posts teaser images are broken
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-13T14:02:37Z" title="Thursday, June 13th 2019, 4:02:37 pm +02:00">Jun 13, 2019</time>_
_Closed <time datetime="2019-06-13T23:58:53Z" title="Friday, June 14th 2019, 1:58:53 am +02:00">Jun 14, 2019</time>_
---

## :bug: Bugreport
Posts teaser images are broken after #809 and #810 

### Steps to reproduce the behavior
1. Visit https://nitro-production.human-connection.org/
2. Login
3. Open Javascript Console
4. See all the 404 error messages

### Expected behavior
Posts images should be visible

### Additional Context

Screenshot - 2019-06-13T155537 115

### PR DESCRIPTION
**Pull request by Tirokk**
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-13T15:03:28Z" title="Thursday, June 13th 2019, 5:03:28 pm +02:00">Jun 13, 2019</time>_
_Merged <time datetime="2019-06-13T23:58:53Z" title="Friday, June 14th 2019, 1:58:53 am +02:00">Jun 14, 2019</time>_
---

Ok, so after I would have to use the same method in three different
locations (`<ds-card>` expects an `image` attribute but cannot render
entire components) I decided to implement the prefix of image urls with
a filter rather than a component. The downside of this is that we have
to add the filter on a lot of component tests. The benefit is less
components and hopefully less complexity.

### Issues

- Fix #820

### CODE REVIEW

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-06-13T15:03:30Z" title="Thursday, June 13th 2019, 5:03:30 pm +02:00">Jun 13, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/821?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@d0fd4de`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

File: webapp/components/Avatar/Avatar.vue [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <template>
-  <ds-avatar :image="avatarUrl" :name="userName" class="avatar" :size="size" />
+  <ds-avatar
+    :image="user && user.avatar | proxyApiUrl"
```
Comment (Tirokk) [inline webapp/components/Avatar/Avatar.vue:3]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-13T23:55:10Z" title="Friday, June 14th 2019, 1:55:10 am +02:00">Jun 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,5 +1,10 @@
 <template>
-  <ds-avatar :image="avatarUrl" :name="userName" class="avatar" :size="size" />
+  <ds-avatar
+    :image="user && user.avatar | proxyApiUrl"
```


I needed to check what the `|` means in this case @roschaefer 
it's a bitwise OR, correct?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#.7c_%28Bitwise_OR%29
can you explain why you used it in this case?

Comment (Tirokk) [reply webapp/components/Avatar/Avatar.vue:3]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-14T07:48:09Z" title="Friday, June 14th 2019, 9:48:09 am +02:00">Jun 14, 2019</time>_

---


No, this is not a bitwise OR. It's the use of a vue filter in a template. See https://vuejs.org/v2/guide/filters.html

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-14T07:49:48Z" title="Friday, June 14th 2019, 9:49:48 am +02:00">Jun 14, 2019</time>_

---


@mattwr18 the only way I tested this was to check the 404 errors in the console. I couldn't find any 404 error with a URL not having the `/api` prefix, so I believe this should be OK. The dump that was used is the production data with URLs pointing to the uploads folder.

**Review by Mogge — APPROVED** [review]
I'm approving this @roschaefer and merging it in as you asked, because if it doesn't work, I'm sure you'll fix it, but I didn't get a chance to pull it down and check it out locally. I'm confident you did though. Would be great if you could post a screen shot, but I'm sure you were busy and overlooked it this time?

Also, I had a doubt, if you don't mind answering for myself and anyone else that has a look :pray: 
:heart:

**Pull request closed without merging**